js/main.js

```diff
@@ -0,0 +1,86 @@
+const DESCRIPTION = [
+  'Красивый закат на берегу моря',
+  'Музей современного искусства',
+  'Театр оперы и балета',
+  'Центральная улица города',
+  'Гроза в деревне'
+];
+
+const MESSAGES = [
+  'Всё отлично!',
+  'В целом всё неплохо. Но не всё.',
+  'Когда вы делаете фотографию, хорошо бы убирать палец из кадра. В конце концов это просто непрофессионально.',
+  'Моя бабушка случайно чихнула с фотоаппаратом в руках и у неё получилась фотография лучше.',
+  'Я поскользнулся на банановой кожуре и уронил фотоаппарат на кота и у меня получилась фотография лучше.',
+  'Лица у людей на фотке перекошены, как будто их избивают. Как можно было поймать такой неудачный момент?!'
+];
+
+const NAMES = [
+  'Антон',
+  'Александр',
+  'Артур',
+  'Зорик',
+  'Влад',
+  'Таня',
+  'Батор'
+];
+  // мин и макс значения id,лайков,аватаров и комментариев;количество объектов в итоговом массиве
+
+const MIN_LIKES = 15;
+const MAX_LIKES = 200;
+const MIN_AVATAR = 1;
+const MAX_AVATAR = 6;
+const MIN_COMMENTS = 0;
+const MAX_COMMENTS = 30;
+const SIMILAR_OBJECTS_COUNT = 25;
+
+// нахождение случайного числа из диапазона и проверка на уникальность
+const generatedNumbers = [];
+const getRandomPositiveInteger = (a, b) => {
+  const lower = Math.ceil(Math.min(a, b));
+  const upper = Math.floor(Math.max(a, b));
+  const result = Math.floor(Math.random() * (upper - lower + 1) + lower);
+  if (generatedNumbers.includes(result)) {
+    return getRandomPositiveInteger(a,b);
+  }
+  generatedNumbers.push(result);
+  return result;
+};
```

Comment:
Эта функция в целом ок. Единтвенное, что в ней можно было бы убрать первые 3 строки и переиспользовать ту, что ниже (где у тебя делается тоже самое). И тогда это бы выглядело вот так:

```suggestion
const generatedNumbers = [];
const getRandomPositiveInteger = (a, b) => {
  const result = getAvatarIndex(a, b);
  if (generatedNumbers.includes(result)) {
    return getRandomPositiveInteger(a,b);
  }
  generatedNumbers.push(result);
  return result;
};
```

Reply:
Но, что более важно, эту функцию можно удалить совсем.
Я не стал специально писать об этом в самом начале, чтобы была возможность потренироваться. 
Но суть задачи у тебя свелась к тому, чтобы сгенерировать уникальный `id` для комментария. Единственное требование в задаче к нему, это уникальность. А значит можно решить задачу простым счётчиком:

```js
let currentCommentId = 1;

const createComment = () => ({
  id: currentCommentId++,
  avatar: 'img/avatar-' + getAvatarIndex(MIN_AVATAR, MAX_AVATAR) + '.svg',
  message: getRandomArrayElement(MESSAGES),
  name: getRandomArrayElement(NAMES)
});
```

Аналогично тому, что ты сделал с объектом